vector/src/main/java/im/vector/app/features/spaces/manage/SpaceManageRoomsViewModel.kt

```diff
@@ -0,0 +1,186 @@
+                setState { copy(currentFilter = action.filter) }
+            }
+            SpaceManageRoomViewAction.ClearSelection        -> {
+                setState { copy(selectedRooms = emptyList()) }
+            }
+            SpaceManageRoomViewAction.BulkRemove            -> {
+                handleBulkRemove()
+            }
+            is SpaceManageRoomViewAction.MarkAllAsSuggested -> {
+                handleBulkMarkAsSuggested(action.suggested)
+            }
+            SpaceManageRoomViewAction.RefreshFromServer     -> {
+                refreshSummaryAPI()
+            }
+        }
+    }
+
+    private fun handleBulkRemove() = withState { state ->
+        setState { copy(actionState = Loading()) }
+        val selection = state.selectedRooms
+        session.coroutineScope.launch(Dispatchers.IO) {
+            val errorList = mutableListOf<Throwable>()
+            selection.forEach {
+                try {
+                    session.spaceService().getSpace(state.spaceId)?.removeChildren(it)
+                } catch (failure: Throwable) {
+                    errorList.add(failure)
+                }
+            }
+            if (errorList.isEmpty()) {
+                // success
+                _viewEvents.post(SpaceManageRoomViewEvents.BulkActionSuccess)
+            } else {
+                _viewEvents.post(SpaceManageRoomViewEvents.BulkActionFailure(errorList.map { errorFormatter.toHumanReadable(it) }))
+            }
+            setState { copy(actionState = Uninitialized) }
+        }
+    }
+
+    private fun handleBulkMarkAsSuggested(suggested: Boolean) = withState { state ->
+        setState { copy(actionState = Loading()) }
+        val selection = state.childrenInfo.invoke()?.filter {
+            state.selectedRooms.contains(it.childRoomId)
+        }.orEmpty()
+        session.coroutineScope.launch(Dispatchers.IO) {
+            val errorList = mutableListOf<Throwable>()
+            selection.forEach { info ->
+                try {
+//                    session.spaceService().getSpace(state.spaceId)?.setChildrenSuggested(roomId, suggested)
+                    session.spaceService().getSpace(state.spaceId)?.addChildren(
+                            roomId = info.childRoomId,
+                            viaServers = info?.viaServers ?: emptyList(),
+                            order = info?.order,
+                            suggested = suggested,
+                            autoJoin = info?.autoJoin ?: false
+                    )
+                } catch (failure: Throwable) {
+                    errorList.add(failure)
+                }
+            }
+            if (errorList.isEmpty()) {
+                // success
+                _viewEvents.post(SpaceManageRoomViewEvents.BulkActionSuccess)
+                refreshSummaryAPI()
+            } else {
+                _viewEvents.post(SpaceManageRoomViewEvents.BulkActionFailure(errorList.map { errorFormatter.toHumanReadable(it) }))
```

Comment:
errorFormatter could be used in the Fragment here. Provide the list of Failure instaed?

Reply:
thx